src/lido/infos.ts

```diff
@@ -0,0 +1,126 @@
+    if (!vaultId.equals(LIDO_ADDRESS)) throw Error(`Error: Lido vaultId must match ${LIDO_ADDRESS.toBase58()}`);
+
+    const vaultAccountInfo = await connection.getAccountInfo(vaultId);
+    if (!vaultAccountInfo) throw Error("Error: Cannot get solido token account");
+
+    const vault: types.VaultInfo = this.parseVault(vaultAccountInfo.data, vaultId);
+
+    return vault;
+  }
+
+  static parseVault(data: Buffer, vaultId: PublicKey): types.VaultInfo {
+    // Decode the Token data using AccountLayout
+    const decodeData = LIDO_LAYOUT.decode(data);
+    const {
+      lidoVersion,
+      manager,
+      stSolMint,
+      exchangeRate,
+      solReserveAuthorityBumpSeed,
+      stakeAuthorityBumpSeed,
+      mintAuthorityBumpSeed,
+      rewardsWithdrawAuthorityBumpSeed,
+      rewardDistribution,
+      feeRecipients,
+      metrics,
+      validators,
+      maintainers,
+    } = decodeData;
+
+    // Ensure the mint matches
+    if (!this._isAllowed(stSolMint)) {
+      throw Error("Error: Not a stSOL token account");
+    }
+
+    return {
+      vaultId,
+      lidoVersion,
+      manager,
+      shareMint: stSolMint,
+      exchangeRate,
+      solReserveAuthorityBumpSeed,
+      stakeAuthorityBumpSeed,
+      mintAuthorityBumpSeed,
+      rewardsWithdrawAuthorityBumpSeed,
+      rewardDistribution,
+      feeRecipients,
+      metrics,
+      validators,
+      maintainers,
+    };
+  }
+
+  static async getAllDepositors(connection: Connection, userKey: PublicKey): Promise<types.DepositorInfo[]> {
+    const { value: accountInfos } = await connection.getTokenAccountsByOwner(userKey, {
+      mint: ST_SOL_MINT_ADDRESS,
+    });
+    const depositors = accountInfos.map((depositor) => this.parseDepositor(depositor.account.data, depositor.pubkey));
+    return depositors;
+  }
+
+  static async getDepositor(connection: Connection, depositorId: PublicKey): Promise<types.DepositorInfo> {
+    const depositorAccountInfo = await connection.getAccountInfo(depositorId);
+    if (!depositorAccountInfo) throw Error("Error: Cannot get depositor account");
+
+    return this.parseDepositor(depositorAccountInfo.data, depositorId);
+  }
+
+  static getDepositorId(vaultId: PublicKey, _userKey: PublicKey, _programId: PublicKey = LIDO_PROGRAM_ID): PublicKey {
+    return vaultId;
+  }
+
+  static parseDepositor(data: Buffer, depositorId: PublicKey): types.DepositorInfo {
+    const decodeData = LIDO_TOKEN_LAYOUT.decode(data);
+    const { mint, amount, owner } = decodeData;
+
+    // Ensure the mint matches
+    if (!this._isAllowed(mint)) {
+      throw Error(`Error: Not a stSOL token account`);
+    }
+
+    return {
+      depositorId,
+      userKey: owner,
+      amount,
+      mint,
+    };
+  }
+
+  private static _isAllowed(mint: PublicKey): boolean {
+    return mint.equals(ST_SOL_MINT_ADDRESS);
+  }
+};
+
+export { infos };
+
+export class VaultInfoWrapper implements IVaultInfoWrapper {
```

Comment:
Please implement `getHeaviestValidator` in VaultInfoWrapper, can checkout the logic inside their SDK [here](https://github.com/lidofinance/solido-sdk/blob/6fadb02977302f73d17bfdaa74ca1479066cd584/src/unstake/getWithdrawInstruction.ts#L17)